lib/widgets/select_input_widget.dart

```diff
@@ -0,0 +1,69 @@
+import 'package:flutter/material.dart';
+import 'package:tech_byte/models/picker_list_item_model.dart';
+import 'package:tech_byte/widgets/picker_list_widget.dart';
+import 'package:tech_byte/widgets/text_input_widget.dart';
+
+class TBSelectInput extends StatefulWidget {
+  final List<TBPickerListItemModel> items;
+  final String label;
+  final void Function(String?) onTap;
+  final String? selectedItem;
+  final bool enabled;
+  final Widget? suffixIcon;
+  final String? suffixText;
+
+  const TBSelectInput(
+      {super.key,
+      required this.label,
+      required this.onTap,
+      this.selectedItem,
+      required this.items,
+      this.enabled = true,
+      this.suffixIcon,
+      this.suffixText});
+
+  @override
+  State<TBSelectInput> createState() => _TBSelectInputState();
+}
+
+class _TBSelectInputState extends State<TBSelectInput> {
+  final TextEditingController _controller = TextEditingController();
+  String? _selectedItem;
+
+  @override
+  void initState() {
+    _selectedItem = widget.selectedItem;
+    _controller.text = _selectedItem ?? "";
+    super.initState();
+  }
+
```

Comment:
You need to dispose TextEditingController, if not controller will not release allocated memory automatically. That memory will be traped and nobody can access it until we kill application.